pkg/services/vulnerability.go

```diff
@@ -438,3 +438,20 @@ func (s *VulnerabilityService) GetWebScanVulnerabilities(ctx context.Context, sc
 
 	return vulns, nil
 }
+
+func (s *VulnerabilityService) GetWebVulnerabilitiesForService(ctx context.Context, serviceID int32) ([]domain.WebVulnerability, error) {
+	_, errService := s.serviceRepo.GetServiceByID(ctx, serviceID)
+	if errService != nil {
+		if errors.Is(errService, sql.ErrNoRows) {
+			return nil, customerrors.ErrServiceNotFound
+		}
+		return nil, errService
+	}
+	vulns, err := s.vulnerRepo.ListWebVulnerabilitiesByServiceID(ctx, serviceID)
+	if err != nil {
+		slog.Debug("Error when retrieving all webvulns with instances of a service", slog.Any("serviceID", serviceID))
+		return nil, err
+	}
```

Comment:
Nice check! Me gusta la idea, sin embargo creo que podemos hacer solo una llamada al repositorio y dejar esto un poco mas granular y eficiente (solo se enfoca en conseguir las web vulns, no el servicio). 


```suggestion
	vulns, err := s.vulnerRepo.ListWebVulnerabilitiesByServiceID(ctx, serviceID)
	if err != nil {
		slog.Error("Error when retrieving all webvulns with instances of a service",
		slog.Any("serviceID", serviceID),
		slog.Any("error", err),
		)
		return nil, err
	}
```

De todas maneras es argumentable... los puntos a evaluar son eficiencia (cantidad de llamadas a la DB), vs distinguir entre un servicio no-existente y uno limpio. 

Considerando que este metodo sera llamado en un metodo mas grande (conseguir las vulnerabilidades NetworkOS y luego las Web para un servicio, yo asumiria que la logica de existencia deberia ir en el metodo grandote. 

Que opinas?

Reply:
Pues en el ticket pusiste que valide lo del servicio y bueno como es joins no se puede verificar eso del servicio no existente, es por ello que lo puse, si bien es cierto esto va ser en uno mas grandote que se puede hacer, pierde creo su independencia de ser reusado. Aunque pensando que solo sera llamado en el grandote, pues si lo puedo cambiar. Lo voy a cambiar para tener menos llamada a bd.